lib/auth/auth_with_roles.go

```diff
@@ -1291,7 +1291,15 @@ func (a *ServerWithRoles) ListUnifiedResources(ctx context.Context, req *proto.L
 		requestableMap: make(map[string]struct{}),
 	}
 
-	for _, kind := range services.UnifiedResourceKinds {
+	// Before doing any listing, verify that the user is allowed to list
+	// at least one of the requested kinds. If no access is permitted, then
+	// return an access denied error.
+	requested := req.Kinds
+	if len(req.Kinds) == 0 {
+		requested = services.UnifiedResourceKinds
+	}
+
+	for _, kind := range requested {
```

Comment:
Update the comment and maybe the capacity where we initialize `kindAccessMap`.

Is there a limit to how many `Kinds` there can be in the request? Should we check if `kindAccessMap[kind]` is already populated before doing anything further with the kind?

Reply:
Will `selectActionChecker`/`checkAction` behave appropriately if a `kind` is not among the ones in `services.UnifiedResourceKinds`? They were skipped as part of normal operation before since we were precomputing access for just the kinds in `services.UnifiedResourceKinds`.